--- src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj	2024-08-26 13:58:55.745998465 -0300
+++ src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj	2024-08-26 13:59:34.439998317 -0300
@@ -560,6 +560,7 @@
     <!-- Try various places to find the runtime. It's either released (use official version),
          public but un-released (use dotnetbuilds/public), or internal and unreleased (use dotnetbuilds/internal) -->
     <ItemGroup>
+      <UrisToDownload Include="file://@@DOWNLOADS_DIR_PATH@@/$(DotNetRuntimeDownloadPath)" />
       <UrisToDownload Include="https://dotnetcli.azureedge.net/dotnet/$(DotNetRuntimeDownloadPath)" />
       <UrisToDownload Include="https://dotnetbuilds.azureedge.net/public/$(DotNetRuntimeDownloadPath)" />
       <UrisToDownload Include="https://dotnetbuilds.azureedge.net/internal/$(DotNetRuntimeDownloadPath)"
